packages/ruleset-migrator/src/tree/index.ts

```diff
@@ -91,14 +94,15 @@ export class Tree {
     );
   }
 
-  public identifier(name: string): namedTypes.Identifier {
-    let uniqName = name;
+  public static identifier(name: string, scope: Scope): namedTypes.Identifier {
+    const baseName = name.replace(/[^$_0-9A-Za-z]/g, '').replace(/^([0-9])/, '_$1');
+    let uniqName = baseName;
     let i = 0;
-    while (this.scope.has(uniqName)) {
-      uniqName = `${name}_${i++}`;
+    while (scope.has(uniqName)) {
+      uniqName = `${baseName}$${i++}`;
```

Comment:
Curious, why don't we scope the entire identifier for imports? Then could just replace invalid characters instead of matching registered words.

```suggestion
      uniqName = `import${i++}_${baseName}`;
```

Reply:
Mostly for the sake of readability.
I try to make the output as clean as possible.
Basically if one has such a function declared
```json
"functions": ["test"]
```
we want to output

```js
import test from '/functions/test.js';
```
This is all for the sake of trying to be as least destructive as we can.